internal/pkg/config/config_test.go

```diff
@@ -3,122 +3,171 @@ package config
+			Version:            "latest",
+			Path:               fixtureSidecarsDir + "/volume-mounts.yaml",
+			EnvCount:           2,
+			ContainerCount:     3,
+			VolumeCount:        2,
+			VolumeMountCount:   1,
+			HostAliasCount:     0,
+			InitContainerCount: 0,
 		},
-		"env1": configExpectation{
-			name:                       "env1",
-			path:                       fixtureSidecarsDir + "/env1.yaml",
-			expectedEnvVarCount:        3,
-			expectedContainerCount:     0,
-			expectedVolumeCount:        0,
-			expectedVolumeMountCount:   0,
-			expectedHostAliasCount:     0,
-			expectedInitContainerCount: 0,
+		"host-aliases": testhelper.ConfigExpectation{
+			Name:               "host-aliases",
+			Version:            "latest",
+			Path:               fixtureSidecarsDir + "/host-aliases.yaml",
+			EnvCount:           2,
+			ContainerCount:     1,
+			VolumeCount:        0,
+			VolumeMountCount:   0,
+			HostAliasCount:     6,
+			InitContainerCount: 0,
 		},
-		"volume-mounts": configExpectation{
-			name:                       "volume-mounts",
-			path:                       fixtureSidecarsDir + "/volume-mounts.yaml",
-			expectedEnvVarCount:        2,
-			expectedContainerCount:     3,
-			expectedVolumeCount:        2,
-			expectedVolumeMountCount:   1,
-			expectedHostAliasCount:     0,
-			expectedInitContainerCount: 0,
+		"init-containers": testhelper.ConfigExpectation{
+			Name:               "init-containers",
+			Version:            "latest",
+			Path:               fixtureSidecarsDir + "/init-containers.yaml",
+			EnvCount:           0,
+			ContainerCount:     2,
+			VolumeCount:        0,
+			VolumeMountCount:   0,
+			HostAliasCount:     0,
+			InitContainerCount: 1,
 		},
-		"host-aliases": configExpectation{
-			name:                       "host-aliases",
-			path:                       fixtureSidecarsDir + "/host-aliases.yaml",
-			expectedEnvVarCount:        2,
-			expectedContainerCount:     1,
-			expectedVolumeCount:        0,
-			expectedVolumeMountCount:   0,
-			expectedHostAliasCount:     6,
-			expectedInitContainerCount: 0,
+		"versioned1": testhelper.ConfigExpectation{
+			Name:               "init-containers",
+			Version:            "v2",
+			Path:               fixtureSidecarsDir + "/init-containers-v2.yaml",
+			EnvCount:           0,
+			ContainerCount:     2,
+			VolumeCount:        0,
+			VolumeMountCount:   0,
+			HostAliasCount:     0,
+			InitContainerCount: 1,
 		},
-		"init-containers": configExpectation{
-			name:                       "init-containers",
-			path:                       fixtureSidecarsDir + "/init-containers.yaml",
-			expectedEnvVarCount:        0,
-			expectedContainerCount:     2,
-			expectedVolumeCount:        0,
-			expectedVolumeMountCount:   0,
-			expectedHostAliasCount:     0,
-			expectedInitContainerCount: 1,
+		// test that a name with spurious use of ":" returns the expected Name and Version
+		"versioned:with:extra:data:v3": testhelper.ConfigExpectation{
```

Comment:
I think  enforcing only one `:` like Docker does will be much cleaner in the long run instead of allowing this and taking the last part as the versions and the first part  as the name:

```
$ docker tag bash:latest bash:v3:latest
Error parsing reference: "bash:v3:latest" is not a valid repository/tag: invalid reference format
````

Reply:
Yea, thats fair. Ill update to check this, cause it definitely makes things cleaner in the long run.